### PR TITLE
Add AsString() to Set

### LINKED ISSUE #36: Empty string is a *rel.genericSet instead of rel.String
Doing
```shell
$ arrai e '""'
{}
```

Internally an empty `*rel.genericSet` can not be converted to `rel.String`

@marcelocantos suggested `Set.AsString() (string, bool)` method

### PR DESCRIPTION
**Pull request by nofun97**
Add AsString() to Set interface which converts any Set to the internal String struct

Fixes #36

### CODE REVIEW

File: rel/value_set.go [hunk 1]
```diff
@@ -327,6 +327,48 @@ func (s *genericSet) Any() Value {
 	panic("Any(): empty set")
 }
 
+func (s *genericSet) AsString() (String, bool) {
+	if s.set.Count() == 0 {
```
Comment (marcelocantos) [inline rel/value_set.go]: `!s.Bool()`

Comment (marcelocantos) [reply rel/value_set.go]: Might be worth changing `Bool` to `IsTrue`. Raise an issue.

Comment (nofun97) [reply rel/value_set.go]: is the purpose of `Bool` to show the `Bool` value of a set or to show if it is not empty? If it's the latter, might be better to change it to `NotEmpty`.

**Comment by nofun97** [comment]
Just remembered I'll have to use the function at the str lib

File: rel/value_set.go (hunk 1 above)
Comment (marcelocantos) [reply rel/value_set.go]: All values have a "truthiness" to them, which determines how they affect logical operations. `0`, `()` and `{}` are all considered "false". Every other value is considered "true". The `Bool()` method expresses this, but `IsTrue` would be less confusing.

**Pull request merged**